learning_resources/views.py

```diff
@@ -376,6 +378,125 @@ class ResourceListItemsViewSet(NestedViewSetMixin, viewsets.ReadOnlyModelViewSet
     ordering = ["position", "-child__last_modified"]
 
 
+@extend_schema(
+    parameters=[
+        OpenApiParameter(
+            name="id",
+            type=OpenApiTypes.INT,
+            location=OpenApiParameter.PATH,
+            description="id of the learning resource",
+        ),
+        OpenApiParameter(
+            name="userlist_id",
+            type=OpenApiTypes.INT,
+            many=True,
+            location=OpenApiParameter.QUERY,
+            description="id of the parent user list",
+        ),
+        OpenApiParameter(
+            name="learning_path_id",
+            type=OpenApiTypes.INT,
+            many=True,
+            location=OpenApiParameter.QUERY,
+            description="id of the parent learning path",
+        ),
+    ],
+)
+class LearningResourceListRelationshipViewSet(viewsets.GenericViewSet):
+    """
+    Viewset for managing relationships between Learning Resources
+    and User Lists / Learning Paths
+    """
+
+    permission_classes = (AnonymousAccessReadonlyPermission,)
+    filter_backends = [MultipleOptionsFilterBackend]
+    queryset = LearningResourceRelationship.objects.select_related("parent", "child")
+    http_method_names = ["patch"]
```

Comment:
*Minor comment:* PUT still seems like a mildly better fit to me since this replaces the existing relationshisps (for that resource). When I briefly tried, I had an issue making PUT work with `detail=false`, but wince you're using `detail=true`, I think it would work.